api/genericresource/parse.go

```diff
@@ -0,0 +1,35 @@
+package genericresource
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/docker/swarmkit/api"
+)
+
+// Parse parses the GenericResource resources given by the arguments
+func Parse(tprs string, resources *api.Resources) error {
+	resources.Generic = make(map[string]*api.GenericResource)
+	mtpr := resources.Generic
+
+	for _, term := range strings.Split(tprs, ";") {
```

Comment:
> --third-party-resource "banana=blue,red,green;apple=8"

Take this example, it doesn't say if a task would take 0.5 apple. I think it should contain resource name, type, and value.